app/models/course/material/folder.rb

```diff
@@ -106,6 +101,16 @@ def initialize_duplicate_children(duplicator, other)
                 map { |folder| duplicator.duplicate(folder) }
   end
 
+  def initialize_duplicate_materials(duplicator, other)
+    self.materials = if other.concrete?
```

Comment:
Is it possible to explain why you introduced this - is it trying to protect the case where we accidentally try to duplicate non-concrete folders? Or are we allowing for duplication of non-concrete folders?

Reply:
@kxmbrian  ```select { |material| duplicator.duplicated?(material) }``` does this means that materials will always be duplicated before folder?  what if the materials are duplicated later?

Reply:
@kxmbrian Can explain a bit more? since the order doesn't matter, why we need to check `duplicator.duplicated?(material)`?